### DISCUSSION TITLE
Why not support creation of empty folders within Transfer?

### DISCUSSION THREAD

**Discussion by stecca21** [aws/aws-sdk-php#2494] [Q&A] [answered]
If I can create an empty "folder" from the AWS management console as well as with putObject why not support the uploading of empty folders within the Transfer operation?

**Comment by yenfryherrerafeliz** [comment, marked as answer]
Hi @stecca21 thanks for opening this discussion. At programming level, what causes that the directories do not get created is the following [condition](https://github.com/aws/aws-sdk-php/blob/4e0edc577543d4c009c1a1b5d70d34f58b9f2e25/src/S3/Transfer.php#L309) where basically the filter will return whatever is not a directory.

So, I will suggest you to please open a feature request [here](https://github.com/aws/aws-sdk-php/issues/new?assignees=&labels=feature-request%2Cneeds-triage&template=feature-request.yml&title=%28short+issue+description%29) for this. I want to mention that we prioritize feature requests based on community reactions and comments.

Thanks!

[a comment by an automated account is not shown]
